nio/util/flags_enum.py

```diff
@@ -91,6 +91,27 @@ def add(self, flag):
             if self._status_change_callback:
                 self._status_change_callback(old_status, self)
 
+    def replace(self, old_flag, new_flag):
+        self._validate_flag(old_flag)
+        self._validate_flag(new_flag)
+
+        # save old status to send along with changed status
+        old_status = copy.deepcopy(self)
+
+        status_changed = False
+        # remove old flag
+        if self._flags[old_flag.name]:
```

Comment:
Won't this raise a `KeyError` if the flag has never been set before? I may be reading the code wrong but I think we probably want to do this instead:
```python
if self._flags.get(old_flag.name, False):
```

Reply:
not really, all flags are assigned during `__init__` when invoking the `clear` method, this is assumed in other places too, I'll add a test to confirm.